src/engine/sparqlExpressions/SparqlExpressionValueGetters.h

```diff
@@ -141,6 +141,32 @@ struct StringValueGetter : Mixin<StringValueGetter> {
   }
 };
 
+// This class can be used as the `ValueGetter` argument of Expression
+// templates. It produces a LiteralOrIri.
+struct LiteralOrIriValueGetter : Mixin<LiteralOrIriValueGetter> {
+  using Mixin<LiteralOrIriValueGetter>::operator();
+
+  std::optional<LiteralOrIri> operator()(ValueId,
+                                         const EvaluationContext*) const;
+
+  std::optional<LiteralOrIri> operator()(const LiteralOrIri& s,
+                                         const EvaluationContext*) const {
+    return s;
+  }
+};
```

Comment:
You also need a second valueGetter which has the semantics of the `onlyReturnXsdStrings`.
(so one value getter for each of your configurations of the configurations of your functions in `ExportQueryExecutionTrees.cpp`.